test/integration/kubernetes_deploy_test.rb

```diff
@@ -1373,7 +1373,7 @@ def test_pvc_no_pv
 
     assert_logs_match_all([
       "Failed to deploy 1 priority resource",
-      "Pod/pvc: TIMED OUT (timeout: 20s)",
+      "Pod/pvc: TIMED OUT (timeout: 10s)",
```

Comment:
This test seems to be missing an assertion about the timeout message. Presumably it is different than in the similar test below?

Reply:
There really isn't much to test for here:

```
[INFO][2019-05-30 14:51:50 -0700]	------------------------------Phase 3: Predeploying priority resources------------------------------
[INFO][2019-05-30 14:51:50 -0700]	Deploying resources:
[INFO][2019-05-30 14:51:50 -0700]	- PersistentVolumeClaim/with-storage-class (timeout: 10s)
[INFO][2019-05-30 14:51:50 -0700]	- PersistentVolumeClaim/without-storage-class (timeout: 300s)
[INFO][2019-05-30 14:51:50 -0700]	Successfully deployed in 0.5s: PersistentVolumeClaim/without-storage-class
[INFO][2019-05-30 14:51:50 -0700]	Continuing to wait for: PersistentVolumeClaim/with-storage-class
[ERROR][2019-05-30 14:52:02 -0700]	PersistentVolumeClaim/with-storage-class rollout timed out after 12.5s
[INFO][2019-05-30 14:52:02 -0700]	
[INFO][2019-05-30 14:52:02 -0700]	------------------------------------------Result: FAILURE-------------------------------------------
[FATAL][2019-05-30 14:52:02 -0700]	Failed to deploy 1 priority resource
[FATAL][2019-05-30 14:52:02 -0700]	
[FATAL][2019-05-30 14:52:02 -0700]	PersistentVolumeClaim/with-storage-class: TIMED OUT (timeout: 10s)
[FATAL][2019-05-30 14:52:02 -0700]	PVC specified a StorageClass of k8s-deploy-test-no-pv-or-sc but the resource does not exist
[FATAL][2019-05-30 14:52:02 -0700]	
[FATAL][2019-05-30 14:52:02 -0700]	  - Final status: Pending
[FATAL][2019-05-30 14:52:02 -0700]	  - Events: None found. Please check your usual logging service (e.g. Splunk).
```

The real error:
```
  status:
    conditions:
    - lastProbeTime: null
      lastTransitionTime: "2019-05-26T00:12:42Z"
      message: '0/1 nodes are available: 1 node(s) didn''t find available persistent
        volumes to bind.'
      reason: Unschedulable
      status: "False"
      type: PodScheduled
    phase: Pending
    qosClass: BestEffort
``` 

Isn't something we log.

Reply:
If I'm understanding this correctly, the message we are printing ("PVC specified a StorageClass of k8s-deploy-test-no-pv-or-sc but the resource does not exist") is misleading--the storage class does exist in this case (unlike in the test below), but there's no PV to satisfy the claim.

Reply:
Crap, I pasted the output for the wrong test (test_pvc_no_pv_or_sc is what's above):
This is the output for test_pvc_no_pv:
```
[INFO][2019-05-31 10:43:34 -0700]	------------------------------Phase 3: Predeploying priority resources------------------------------
[INFO][2019-05-31 10:43:34 -0700]	Deploying resources:
[INFO][2019-05-31 10:43:34 -0700]	- PersistentVolumeClaim/with-storage-class (timeout: 10s)
[INFO][2019-05-31 10:43:34 -0700]	- PersistentVolumeClaim/without-storage-class (timeout: 300s)
[INFO][2019-05-31 10:43:35 -0700]	Successfully deployed in 0.5s: PersistentVolumeClaim/with-storage-class
[INFO][2019-05-31 10:43:35 -0700]	Continuing to wait for: PersistentVolumeClaim/without-storage-class
[INFO][2019-05-31 10:43:38 -0700]	Successfully deployed in 3.5s: PersistentVolumeClaim/without-storage-class
[INFO][2019-05-31 10:43:38 -0700]	
[INFO][2019-05-31 10:43:38 -0700]	Deploying Pod/pvc (timeout: 10s)
[INFO][2019-05-31 10:43:38 -0700]	Streaming logs from Pod/pvc container 'hello-cloud':
[ERROR][2019-05-31 10:43:50 -0700]	Pod/pvc rollout timed out after 12.4s
[INFO][2019-05-31 10:43:50 -0700]	
[INFO][2019-05-31 10:43:50 -0700]	------------------------------------------Result: FAILURE-------------------------------------------
[FATAL][2019-05-31 10:43:50 -0700]	Failed to deploy 1 priority resource
[FATAL][2019-05-31 10:43:50 -0700]	
[FATAL][2019-05-31 10:43:50 -0700]	Pod/pvc: TIMED OUT (timeout: 10s)
[FATAL][2019-05-31 10:43:50 -0700]	Kubernetes will continue to attempt to deploy this resource in the cluster, but at this point it is considered unlikely that it will succeed.
[FATAL][2019-05-31 10:43:50 -0700]	If you have reason to believe it will succeed, retry the deploy to continue to monitor the rollout.
[FATAL][2019-05-31 10:43:50 -0700]	
[FATAL][2019-05-31 10:43:50 -0700]	  - Final status: Pending
[FATAL][2019-05-31 10:43:50 -0700]	  - Events: None found. Please check your usual logging service (e.g. Splunk).
WARNING: Skipping logging assertions while logs are redirected to stderr
```

Reply:
Can you paste the corresponding status too? Seems like we're not being very helpful here so I'm wondering if there's something more we could say.

Reply:
The status above is correct for this test. Sorry about the confusion.

Reply:
Oh, I see. So in the case where the user deploys an impossible PVC making a claim on a non-existent PV, we show the PVC as successful and any dependent pod as failing? It seems odd that the PVC would succeed in this case.

Reply:
From memory this test was written to show this behavior. It happens because the PVC uses a storage class with `volumeBindingMode: WaitForFirstConsumer`. This is a mode where you can 🔫 👣 . This is all we get for the PVC status: 

```
status:
  phase: Pending
```

which is can be a reasonable state w/ this mode.

Reply:
Adding a bit more logging:
```
[INFO][2019-06-03 10:23:31 -0700]	
[INFO][2019-06-03 10:23:31 -0700]	------------------------------Phase 3: Predeploying priority resources------------------------------
[INFO][2019-06-03 10:23:31 -0700]	Deploying resources:
[INFO][2019-06-03 10:23:31 -0700]	- PersistentVolumeClaim/with-storage-class (timeout: 10s)
[INFO][2019-06-03 10:23:31 -0700]	- PersistentVolumeClaim/without-storage-class (timeout: 300s)
[INFO][2019-06-03 10:23:32 -0700]	Successfully deployed in 0.5s: PersistentVolumeClaim/with-storage-class
[INFO][2019-06-03 10:23:32 -0700]	Continuing to wait for: PersistentVolumeClaim/without-storage-class
[INFO][2019-06-03 10:23:35 -0700]	Successfully deployed in 3.5s: PersistentVolumeClaim/without-storage-class
[INFO][2019-06-03 10:23:35 -0700]	
[INFO][2019-06-03 10:23:35 -0700]	Deploying Pod/pvc (timeout: 10s)
[INFO][2019-06-03 10:23:35 -0700]	Streaming logs from Pod/pvc container 'hello-cloud':
[ERROR][2019-06-03 10:23:47 -0700]	Pod/pvc rollout timed out after 12.5s
[INFO][2019-06-03 10:23:47 -0700]	
[INFO][2019-06-03 10:23:47 -0700]	------------------------------------------Result: FAILURE-------------------------------------------
[FATAL][2019-06-03 10:23:47 -0700]	Failed to deploy 1 priority resource
[FATAL][2019-06-03 10:23:47 -0700]	
[FATAL][2019-06-03 10:23:47 -0700]	Pod/pvc: TIMED OUT (timeout: 10s)
[FATAL][2019-06-03 10:23:47 -0700]	Pod could not be scheduled because 0/1 nodes are available: 1 node(s) didn't find available persistent volumes to bind.
[FATAL][2019-06-03 10:23:47 -0700]	
[FATAL][2019-06-03 10:23:47 -0700]	  - Final status: Pending
[FATAL][2019-06-03 10:23:47 -0700]	  - Events: None found. Please check your usual logging service (e.g. Splunk).
```